src/modules/core/commands/confess.ts

```diff
@@ -58,17 +59,37 @@ export default Command(
   }
 );
 
-const isImageUrl = (text: string): boolean => {
+const tenorRegex = /https:\/\/tenor.com\/.*\/view\/.*-[0-9]+/;
+const getImageUrl = async (text: string): Promise<string | undefined> => {
+  if (text.startsWith("https://tenor.com") && text.match(tenorRegex)) {
+    // find out gif name from URL
+    // last part of the URL
+    let gifName = text.split("/").pop()!;
```

Comment:
I think this would be better if you moved this further down and used the URL object to get the `url.host`, which you can check with `url.host === "tenor.com"`, and `urlPath.base`, from https://nodejs.org/api/path.html#pathparsepath, which would give you the last part of the url, so tenor.com/a/b/c would give you `c`; although I think the tenorRegex gets to stay anyways

Reply:
done